src/transformers/configuration_utils.py

```diff
@@ -467,7 +467,43 @@ def save_pretrained(self, save_directory: Union[str, os.PathLike], push_to_hub:
             )
 
     @classmethod
-    def from_pretrained(cls, pretrained_model_name_or_path: Union[str, os.PathLike], **kwargs) -> "PretrainedConfig":
+    def _set_token_in_kwargs(self, kwargs, token=None):
+        """Temporary method to deal with `token` and `use_auth_token`.
+
+        This method is to avoid apply the same changes in all model config classes that overwrite `from_pretrained`.
+
+        Need to clean up `use_auth_token` in a follow PR.
+        """
+        # Some model config classes like CLIP define their own `from_pretrained` without the new argument `token` yet.
+        if token is None:
+            token = kwargs.pop("token", None)
+        use_auth_token = kwargs.pop("use_auth_token", None)
+
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed in v5 of Transformers.", FutureWarning
+            )
+            if token is not None:
+                raise ValueError(
+                    "`token` and `use_auth_token` are both specified. Please set only the argument `token`."
+                )
+            token = use_auth_token
+
+        if token is not None:
+            # change to `token` in a follow-up PR
+            kwargs["use_auth_token"] = token
```

Comment:
We have quite some model classes (e.g. clip-like family) whose config classes have their own `from_pretrained`.

This private `_set_token_in_kwargs` method is to make life easier when dealing with `token` and `use_auth_token`.

We will see what the best way is in a follow up PR when we want to make those customized `from_pretrained` with explicit arguments.

Reply:
@sgugger Now it's the good time if you are motivated after 🦷.

Reply:
Ugh... Annoying! This fix works in the meantime.